src/main/java/strategy/impl/OperationStrategyImpl.java

```diff
@@ -0,0 +1,20 @@
+package strategy.impl;
+
+import java.util.Map;
+import model.FruitTransaction;
+import strategy.OperationHandler;
+import strategy.OperationStrategy;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private Map<FruitTransaction.Operation, OperationHandler> operationHandlers;
+
+    public OperationStrategyImpl(Map<FruitTransaction.Operation,
+            OperationHandler> operationHandlers) {
+        this.operationHandlers = operationHandlers;
+    }
+
+    @Override
+    public OperationHandler getOperationHandler(FruitTransaction.Operation operation) {
+        return operationHandlers.get(operation);
```

Comment:
There is a potential issue with the `getOperationHandler` method. It will return `null` if there is no corresponding operation handler in the map. It might be a good idea to handle this case explicitly, either by throwing an exception or by providing a default operation handler.